twoliter/src/lock.rs

```diff
@@ -139,15 +331,98 @@ impl Lock {
         Ok(lock)
     }
 
+    fn external_kit_metadata(&self) -> ExternalKitMetadata {
+        ExternalKitMetadata {
+            sdk: self.sdk.clone(),
+            kits: self.kit.clone(),
+        }
+    }
+
+    /// Fetches all external kits defined in a Twoliter.lock to the build directory
+    pub(crate) async fn fetch(&self, project: &Project, arch: &str) -> Result<()> {
+        let target_dir = project.external_kits_dir();
+        create_dir_all(&target_dir).await.context(format!(
+            "failed to create external-kits directory at {}",
+            target_dir.display()
+        ))?;
+        for image in self.kit.iter() {
+            self.extract_kit(&project.external_kits_dir(), image, arch)
+                .await?;
+        }
+        let mut kit_list = Vec::new();
+        let mut ser =
+            serde_json::Serializer::with_formatter(&mut kit_list, CanonicalJsonFormatter::new());
+        self.external_kit_metadata()
+            .serialize(&mut ser)
+            .context("failed to serialize external kit metadata")?;
+        write(project.external_kits_metadata(), kit_list.as_slice())
+            .await
+            .context(format!(
+                "failed to write external kit metadata: {}",
+                project.external_kits_metadata().display()
+            ))?;
+
+        Ok(())
+    }
+
+    async fn get_manifest(&self, image: &LockedImage, arch: &str) -> Result<ManifestView> {
+        let manifest_bytes = docker!(
+            ["manifest", "inspect", image.source.as_str()],
+            format!("failed to find a kit {}", image.to_string())
+        );
+        let manifest_list: ManifestListView = serde_json::from_slice(manifest_bytes.as_slice())
+            .context("failed to deserialize manifest list")?;
+        let docker_arch = DockerArchitecture::try_from(arch)?;
+        manifest_list
+            .manifests
+            .iter()
+            .find(|x| x.platform.as_ref().unwrap().architecture == docker_arch)
+            .cloned()
+            .context(format!(
+                "could not find kit image for architecture '{}' at {}",
+                docker_arch, image.source
+            ))
+    }
+
+    async fn extract_kit<P>(&self, path: P, image: &LockedImage, arch: &str) -> Result<()>
+    where
+        P: AsRef<Path>,
+    {
+        let vendor = image.vendor.clone();
+        let name = image.name.clone();
+        let docker_arch = match arch {
+            "x86_64" => Ok("amd64"),
+            "aarch64" => Ok("arm64"),
+            _ => Err(anyhow!("unsupported architecture '{}'", arch)),
+        }?;
+        let target_path = path.as_ref().join(format!("{vendor}/{name}/{arch}"));
+        let cache_path = path.as_ref().join("cache");
+        create_dir_all(&target_path).await?;
+        create_dir_all(&cache_path).await?;
+
+        // First get the manifest for the specific requested architecture
+        let manifest = self.get_manifest(image, docker_arch).await?;
```

Comment:
can get rid of `let docker_arch = ...` above and change this:
```suggestion
        // First get the manifest for the specific requested architecture
        let manifest = self.get_manifest(image, arch).await?;
```